projects/packages/jetpack-mu-wpcom/src/features/wpcom-global-styles/index.php

```diff
@@ -783,6 +783,67 @@ function wpcom_global_styles_is_previewing_premium_theme_without_premium_plan( $
 	return ! $has_premium_plan_or_higher;
 }
 
+/**
+ * Returns the experiment variation. The code is bad, it's duplicated. It's a tradeoff since this code will be deleted
+ * after the experiment and it's easier to duplicate than to refactor the original given testing complexities and time constraints.
+ * Yes, this is inefficient and duplicates a lot of code, I know.
+ *
+ * @return bool Whether the site has access to Global Styles with a Personal plan.
+ */
+function get_global_styles_on_personal_variation() {
+	// Environment guard: only WP.com or Atomic.
+	$is_wpcom  = defined( 'IS_WPCOM' ) && IS_WPCOM;
+	$is_atomic = defined( 'IS_ATOMIC' ) && IS_ATOMIC;
+	if ( ! $is_wpcom && ! $is_atomic ) {
+		return false;
+	}
+
+	$wpcom_blog_owner_id = null;
+	$wpcom_blog_id       = get_wpcom_blog_id();
+	if ( function_exists( 'wpcom_get_blog_owner' ) ) {
+		$wpcom_blog_owner_id = wpcom_get_blog_owner( $wpcom_blog_id );
+	}
+
+	if ( ! $wpcom_blog_id ) {
+		return false;
+	}
+
+	$experiment_key = 'calypso_plans_global_styles_personal_20251124_v5';
+	$cache_group    = 'a8c_experiments';
+	$cache_key      = sprintf(
+		'global-styles-personal-variation-%d',
+		$wpcom_blog_id
+	);
+
+	// Cache lookup.
+	$found     = false;
+	$variation = wp_cache_get( $cache_key, $cache_group, false, $found );
+	$found     = apply_filters( 'wpcom_global_styles_experiment_cache', $found );
+	if ( true === $found ) {
+		return $variation;
+	}
+
+	if ( $is_atomic ) {
+		// Atomic: ask WP.com assignment API for this user.
+		$variation = wpcom_global_styles_assignment_api_request( $wpcom_blog_id );
```

Comment:
`wpcom_global_styles_assignment_api_request` does not return the variation, but a boolean. It should be updated to return the variation directly.